crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -0,0 +1,451 @@
+        {
+            let mut cache = self.blob_cache.lock();
+            for tx in txs {
+                if let Some(blob) = cache.get(&tx) {
+                    res.push((tx, blob.clone()));
+                } else {
+                    cache_miss.push(tx)
+                }
+            }
+        }
+        if cache_miss.is_empty() {
+            return Ok(res)
+        }
+        let from_disk = self.read_many_decoded(cache_miss);
+        if from_disk.is_empty() {
+            return Ok(res)
+        }
+        let mut cache = self.blob_cache.lock();
+        for (tx, data) in from_disk {
+            cache.insert(tx, data.clone());
+            res.push((tx, data));
+        }
+
+        Ok(res)
+    }
+
+    #[inline]
+    fn get_exact(&self, txs: Vec<B256>) -> Result<Vec<BlobTransactionSidecar>, BlobStoreError> {
+        let mut res = Vec::with_capacity(txs.len());
+        for tx in txs {
+            let blob = self.get_one(tx)?.ok_or_else(|| BlobStoreError::MissingSidecar(tx))?;
+            res.push(blob)
+        }
+
+        Ok(res)
+    }
+}
+
+impl fmt::Debug for DiskFileBlobStoreInner {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("DiskFileBlobStoreInner")
+            .field("blob_dir", &self.blob_dir)
+            .field("cached_blobs", &self.blob_cache.try_lock().map(|lock| lock.len()))
+            .finish()
+    }
+}
+
+/// Errors that can occur when interacting with a disk file blob store.
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum DiskFileBlobStoreError {
+    /// Thrown during [DiskFileBlobStore::open] if the blob store directory cannot be opened.
+    #[error("failed to open blobstore at {0}: {1}")]
+    Open(PathBuf, io::Error),
+    #[error("[{0}] failed to read blob file at {1}: {2}")]
+    ReadFile(TxHash, PathBuf, io::Error),
+    #[error("[{0}] failed to write blob file at {1}: {2}")]
+    WriteFile(TxHash, PathBuf, io::Error),
+    #[error("[{0}] failed to delete blob file at {1}: {2}")]
+    DeleteFile(TxHash, PathBuf, io::Error),
+}
+
+impl From<DiskFileBlobStoreError> for BlobStoreError {
+    fn from(value: DiskFileBlobStoreError) -> Self {
+        BlobStoreError::Other(Box::new(value))
+    }
+}
+
+/// Configuration for a disk file blob store.
+#[derive(Debug, Clone)]
+pub struct DiskFileBlobStoreConfig {
+    /// The maximum number of blobs to keep in the in memory blob cache.
+    pub max_cached_entries: u32,
+    /// How to open the blob store.
+    pub open: OpenDiskFileBlobStore,
+}
+
+impl Default for DiskFileBlobStoreConfig {
+    fn default() -> Self {
+        Self { max_cached_entries: DEFAULT_MAX_CACHED_BLOBS, open: Default::default() }
+    }
+}
+
+/// How to open a disk file blob store.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum OpenDiskFileBlobStore {
+    /// Clear everything in the blob store.
+    #[default]
+    Clear,
+    /// Keep the existing blob store and index
+    ReIndex,
+}
```

Comment:
guessing this part will be implemented later because this part of the config is ignored when the store is created

Reply:
yeah, since we don't retire pooled txs to disk, this is redundant atm, just a placeholder